src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs

```diff
@@ -9724,5 +9724,86 @@ void Main()
         var x = [|{keyword}|];
     }}");
         }
+
+        [WorkItem(60842, "https://github.com/dotnet/roslyn/issues/60842")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateVariable)]
+        public async Task TestGenerateParameterInAsycMethodBeforeCancellationToken_OneParameter()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System.Threading;
+using System.Threading.Tasks;
+
+class C
+{
+    public async Task M(CancellationToken cancellationToken)
+    {
+        await Task.Delay([|time|]);
+    }
+}",
+@"using System.Threading;
+using System.Threading.Tasks;
+
+class C
+{
+    public async Task M(System.TimeSpan time, CancellationToken cancellationToken)
+    {
+        await Task.Delay(time);
+    }
+}", index: 4);
+        }
+
+        [WorkItem(60842, "https://github.com/dotnet/roslyn/issues/60842")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateVariable)]
+        public async Task TestGenerateParameterInAsycMethodBeforeCancellationToken_SeveralParameters()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System.Threading;
+using System.Threading.Tasks;
+
+class C
+{
+    public async Task M(string someParameter, CancellationToken cancellationToken)
+    {
+        await Task.Delay([|time|]);
+    }
+}",
+@"using System.Threading;
+using System.Threading.Tasks;
+
+class C
+{
+    public async Task M(string someParameter, System.TimeSpan time, CancellationToken cancellationToken)
+    {
+        await Task.Delay(time);
+    }
+}", index: 4);
+        }
+
+        [WorkItem(60842, "https://github.com/dotnet/roslyn/issues/60842")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateVariable)]
+        public async Task TestGenerateParameterNotBeforeCancellationTokenIfMethodIsNotAsync()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System.Threading;
+using System.Threading.Tasks;
+
+class C
+{
+    public Task M(CancellationToken cancellationToken)
+    {
+        await Task.Delay([|time|]);
+    }
+}",
+@"using System.Threading;
+using System.Threading.Tasks;
+
+class C
+{
+    public Task M(CancellationToken cancellationToken, System.TimeSpan time)
```

Comment:
CT should still go last even if not async.

Reply:
That was proposed by the issue